Tests/Realm.Tests/Sync/SyncMigrationTests.cs

```diff
@@ -0,0 +1,365 @@
+
+                Assert.That(objv1.BoolValue, Is.EqualTo(true));
+                Assert.That(objv1.DateValue, Is.EqualTo(date));
+                Assert.That(objv1.DecimalValue, Is.EqualTo(new Decimal128(324.987m)));
+                Assert.That(objv1.DoubleValue, Is.EqualTo(-999.87654321));
+                Assert.That(objv1.IntValue, Is.EqualTo(42));
+                Assert.That(objv1.ObjectIdValue, Is.EqualTo(oid));
+                Assert.That(objv1.StringValue, Is.EqualTo("bla bla"));
+                Assert.That(objv1.UuidValue, Is.EqualTo(uuid));
+                Assert.That(objv1.BinaryValue, Is.EqualTo(binary));
+                Assert.That(objv1.WillBeRemoved, Is.EqualTo(string.Empty));
+
+                var realmv2 = await OpenRealm(differentiator, typeof(NullablesV0), schemaVersion: 2);
+                var objv2 = realmv2.All<NullablesV0>().Single();
+
+                Assert.That(objv2.BoolValue, Is.EqualTo(true));
+                Assert.That(objv2.DateValue, Is.EqualTo(date));
+                Assert.That(objv2.DecimalValue, Is.EqualTo(new Decimal128(324.987m)));
+                Assert.That(objv2.DoubleValue, Is.EqualTo(-999.87654321));
+                Assert.That(objv2.IntValue, Is.EqualTo(42));
+                Assert.That(objv2.ObjectIdValue, Is.EqualTo(oid));
+                Assert.That(objv2.StringValue, Is.EqualTo("bla bla"));
+                Assert.That(objv2.UuidValue, Is.EqualTo(uuid));
+                Assert.That(objv2.BinaryValue, Is.EqualTo(binary));
+
+                realmv0.Write(() =>
+                {
+                    objv0.BoolValue = null;
+                    objv0.DateValue = null;
+                    objv0.DecimalValue = null;
+                    objv0.DoubleValue = null;
+                    objv0.IntValue = null;
+                    objv0.ObjectIdValue = null;
+                    objv0.StringValue = null;
+                    objv0.UuidValue = null;
+                    objv0.BinaryValue = null;
+                });
+
+                await WaitForUploadAsync(realmv0);
+                await WaitForDownloadAsync(realmv1);
+                await WaitForDownloadAsync(realmv2);
+
+                Assert.That(objv1.BoolValue, Is.EqualTo(false));
+                Assert.That(objv1.DateValue, Is.EqualTo(new DateTimeOffset(1, 1, 1, 0, 0, 0, TimeSpan.Zero)));
+                Assert.That(objv1.DecimalValue, Is.EqualTo(Decimal128.Zero));
+                Assert.That(objv1.DoubleValue, Is.EqualTo(0));
+                Assert.That(objv1.IntValue, Is.EqualTo(0));
+                Assert.That(objv1.ObjectIdValue, Is.EqualTo(ObjectId.Empty));
+                Assert.That(objv1.StringValue, Is.EqualTo(string.Empty));
+                Assert.That(objv1.UuidValue, Is.EqualTo(Guid.Empty));
+                Assert.That(objv1.BinaryValue, Is.EqualTo(Array.Empty<byte>()));
+                Assert.That(objv1.WillBeRemoved, Is.EqualTo(string.Empty));
```

Comment:
So the server is keeping different property values depending on the schema version?

Reply:
Makes sense, I read it wrong